legacy/preferences/src/main/java/app/k9mail/legacy/preferences/DefaultSettingsChangeBroker.kt

```diff
@@ -0,0 +1,26 @@
+package app.k9mail.legacy.preferences
+
+class DefaultSettingsChangeBroker(
+    private val subscribers: MutableSet<SettingsChangeSubscriber> = mutableSetOf(),
+) : SettingsChangeBroker, SettingsChangePublisher {
+
+    private val lock = Any()
+
+    override fun subscribe(subscriber: SettingsChangeSubscriber) {
+        synchronized(lock) {
+            subscribers.add(subscriber)
+        }
+    }
+
+    override fun unsubscribe(subscriber: SettingsChangeSubscriber) {
+        synchronized(lock) {
+            subscribers.remove(subscriber)
+        }
+    }
+
+    override fun publish() {
+        for (subscriber in subscribers) {
+            subscriber.receive()
+        }
```

Comment:
This accesses `subscribers` without synchronizing on `lock`. If a subscriber is removed in one thread while this is iterating over subscribers in another thread, a `ConcurrentModificationException` could be thrown.

The problem can be avoided by also wrapping this method body in `synchronized(lock) { … }` or by using `CopyOnWriteArraySet` which can be used without any manual locking/synchronization.

Reply:
I avoided synchronizing this method to balance performance with the low likelihood of concurrent modifications in our use case. However, I agree that the current approach is not robust and should be replaced with a more comprehensive solution in the future.

The `CopyOnWriteArraySet` is a Java only dependency and I opted for a Kotlin specific alternative.

Reply:
I think it's enough to add `@Volatile` to the set of subscribers to prevent `ConcurrentModificationException`.

Reply:
I added a lock around the receivers to avoid `ConcurrentModificationException`